pkg/storegateway/series_chunks.go

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package storegateway
+
+// seriesChunksSetIterator is the interface implemented by an iterator returning a sequence of seriesChunksSet.
+//
+//nolint:unused // dead code while we are working on PR 3355
+type seriesChunksSetIterator interface {
+	Next() bool
+	At() seriesChunksSet
+	Err() error
+}
+
+// seriesChunksSet holds a set of series, each with its own chunks.
+//
+//nolint:unused // dead code while we are working on PR 3355
+type seriesChunksSet struct {
+	series []seriesEntry // this should ideally be its own type that doesn't have the refs
```

Comment:
I would remove this comment. This is a TODO for us, but I wouldn't upstream it.
```suggestion
	series []seriesEntry
```